diff --git a/src/fdo.cpp b/src/fdo.cpp
index 4178384..20c4d9f 100644
--- a/src/fdo.cpp
+++ b/src/fdo.cpp
@@ -27,23 +27,26 @@
 #include <cstring>
 #include <wpe/wpe.h>
 
+static void* loadObject(const char* object_name)
+{
+    if (!std::strcmp(object_name, "_wpe_renderer_host_interface"))
+        return &fdo_renderer_host;
+
+    if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_interface"))
+        return &fdo_renderer_backend_egl;
+    if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_target_interface"))
+        return &fdo_renderer_backend_egl_target;
+    if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_offscreen_target_interface"))
+        return &fdo_renderer_backend_egl_offscreen_target;
+
+    return nullptr;
+}
+
 extern "C" {
 
 __attribute__((visibility("default")))
 struct wpe_loader_interface _wpe_loader_interface = {
-    [](const char* object_name) -> void* {
-        if (!std::strcmp(object_name, "_wpe_renderer_host_interface"))
-            return &fdo_renderer_host;
-
-        if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_interface"))
-            return &fdo_renderer_backend_egl;
-        if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_target_interface"))
-            return &fdo_renderer_backend_egl_target;
-        if (!std::strcmp(object_name, "_wpe_renderer_backend_egl_offscreen_target_interface"))
-            return &fdo_renderer_backend_egl_offscreen_target;
-
-        return nullptr;
-    },
+    loadObject,
 };
 
 }
